From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tri Vo <trong@google.com>
Date: Fri, 10 May 2019 14:33:29 -0700
Subject: ANDROID: gki_defconfig: Remove cuttlefish-specific configs

Left enabled:
- configs for virtualization CONFIG_PARAVIRT, CONFIG_VIRTIO_*, etc.
- cuttlefish doesn't boot without CONFIG_KSM, CONFIG_SGETMASK_SYSCALL,
  CONFIG_SCHED_AUTOGROUP

A number of config should be modules, but due to lack of module support
in userspace are left built in:
- CONFIG_PCI_HOST_GENERIC
- CONFIG_SERIAL_*
- CONFIG_MAC80211
- CONFIG_SND_INTEL8X0
- CONFIG_RTC_DRV_PL030
- CONFIG_RTC_DRV_PL031
- CONFIG_CRYPTO_ADIANTUM
- CONFIG_CRYPTO_ZSTD

Test: boot cuttlefish on arm64
Signed-off-by: Tri Vo <trong@google.com>
(partially cherry picked from commit a651d6eccc912ea60d8b79ddeec23ffb1f85b65f)
---
 arch/arm64/configs/gki_defconfig | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index a747e222db8e..41a892ca324d 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -34,7 +34,6 @@ CONFIG_EMBEDDED=y
 # CONFIG_SLAB_MERGE_DEFAULT is not set
 CONFIG_PROFILING=y
 CONFIG_SCHED_MC=y
-CONFIG_HZ_100=y
 CONFIG_SECCOMP=y
 CONFIG_PARAVIRT=y
 CONFIG_ARMV8_DEPRECATED=y
@@ -56,7 +55,6 @@ CONFIG_CPU_FREQ_TIMES=y
 CONFIG_CPU_FREQ_DEFAULT_GOV_SCHEDUTIL=y
 CONFIG_CPU_FREQ_GOV_POWERSAVE=y
 CONFIG_CPU_FREQ_GOV_CONSERVATIVE=y
-CONFIG_CPUFREQ_DT=y
 CONFIG_ARM_SCPI_CPUFREQ=y
 CONFIG_ARM_SCMI_CPUFREQ=y
 CONFIG_ARM_SCMI_PROTOCOL=y
@@ -90,10 +88,6 @@ CONFIG_INET_ESP=y
 # CONFIG_INET_XFRM_MODE_BEET is not set
 CONFIG_INET_UDP_DIAG=y
 CONFIG_INET_DIAG_DESTROY=y
-CONFIG_TCP_CONG_ADVANCED=y
-# CONFIG_TCP_CONG_BIC is not set
-# CONFIG_TCP_CONG_WESTWOOD is not set
-# CONFIG_TCP_CONG_HTCP is not set
 CONFIG_IPV6_ROUTER_PREF=y
 CONFIG_IPV6_ROUTE_INFO=y
 CONFIG_IPV6_OPTIMISTIC_DAD=y
@@ -191,10 +185,8 @@ CONFIG_RFKILL=y
 CONFIG_PCI=y
 CONFIG_PCI_HOST_GENERIC=y
 # CONFIG_UEVENT_HELPER is not set
-CONFIG_DEVTMPFS=y
 # CONFIG_ALLOW_DEV_COREDUMP is not set
 CONFIG_DEBUG_DEVRES=y
-CONFIG_OF_UNITTEST=y
 CONFIG_ZRAM=y
 CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_RAM=y
@@ -214,8 +206,6 @@ CONFIG_DM_VERITY_AVB=y
 CONFIG_DM_VERITY_FEC=y
 CONFIG_DM_BOW=y
 CONFIG_NETDEVICES=y
-CONFIG_NETCONSOLE=y
-CONFIG_NETCONSOLE_DYNAMIC=y
 CONFIG_TUN=y
 CONFIG_VIRTIO_NET=y
 # CONFIG_ETHERNET is not set
@@ -366,7 +356,6 @@ CONFIG_XZ_DEC=y
 CONFIG_PRINTK_TIME=y
 CONFIG_DEBUG_INFO=y
 # CONFIG_ENABLE_MUST_CHECK is not set
-CONFIG_FRAME_WARN=1024
 # CONFIG_SECTION_MISMATCH_WARN_ONLY is not set
 CONFIG_MAGIC_SYSRQ=y
 CONFIG_DEBUG_STACK_USAGE=y
@@ -375,5 +364,4 @@ CONFIG_SOFTLOCKUP_DETECTOR=y
 # CONFIG_DETECT_HUNG_TASK is not set
 CONFIG_PANIC_TIMEOUT=5
 CONFIG_SCHEDSTATS=y
-CONFIG_RCU_CPU_STALL_TIMEOUT=60
 # CONFIG_RUNTIME_TESTING_MENU is not set
